feature_layers/display-clusters/src/main/java/com/esri/samples/display_clusters/DisplayClustersSample.java

```diff
@@ -89,46 +103,40 @@ public void start(Stage stage) {
               return;
             }
 
-            mapView.getCallout().setVisible(false);
+            mapView.getCallout().dismiss();
             Point2D point = new Point2D(mouseEvent.getX(), mouseEvent.getY());
-            ListenableFuture<IdentifyLayerResult> identifiedLayerResults = mapView.identifyLayerAsync(powerPlantsLayer, point, 3.0, false);
-            identifiedLayerResults.addDoneListener(() -> {
+            ListenableFuture<IdentifyLayerResult> identifiedLayerResultsFuture = mapView.identifyLayerAsync(powerPlantsLayer, point, 3.0, false);
+            identifiedLayerResultsFuture.addDoneListener(() -> {
+              IdentifyLayerResult identifiedLayerResults = null;
               try {
-                if (identifiedLayerResults.get() == null || identifiedLayerResults.get().getPopups().isEmpty()) {
+                identifiedLayerResults = identifiedLayerResultsFuture.get(10, TimeUnit.SECONDS);
+                if (identifiedLayerResults == null || identifiedLayerResults.getPopups().isEmpty()) {
                   return;
                 }
 
-                Popup popup = identifiedLayerResults.get().getPopups().get(0);
+                Popup popup = identifiedLayerResults.getPopups().get(0);
 
-                Callout callout = mapView.getCallout();
+                callout = mapView.getCallout();
                 String htmlText = popup.getDescription();
 
-                WebView webView = new WebView();
-                VBox region = new VBox();
-                WebEngine webEngine;
-                webEngine = webView.getEngine();
                 webEngine.loadContent(htmlText);
-                webView.setMaxHeight(100);
-                webView.setMaxWidth(150);
-                region.getChildren().add(webView);
 
                 callout.setCustomView(region);
```

Comment:
I think the callout returned by the `MapView` is always the exact same object so the `setCustomView` can be done once when the region is created.

Reply:
Good point, I moved both `callout = mapView.getCallout();` and `callout.setCustomView(region);` outside where the pane is created.